### DISCUSSION TITLE
Can the java version chaincode deploy index?

### DISCUSSION THREAD

**Discussion by csb123452010** [hyperledger/fabric-chaincode-java#328] [Q&A]
When I deploy the java version asset-transfer-private-data chaincode in test-network, the index deployment was not successful. And the go version asset-transfer-private-data chaincode is OK.

Then, I want check the reason, but there are no java-version asset-transfer-ledger-quires chaincode. So I copied the index from go-version asset-transfer-ledger-quires chaincode to go-version asset-transfer-basic chaincode and java-version asset-transfer-basic chaincode. And I changed the field in the index.

The go-version asset-transfer-basic chaincode with index is OK. However, the java-version asset-transfer-basic chaincode with index was not successful as well.

**Comment by bestbeforetoday** [comment]
Couch DB indexes _should_ work with Java chaincode in the same way as other chaincode languages. Make sure that the packaged chaincode contains the index in the correct location, relative to the root of the chaincode, not down inside any source folders. See these pages in the Fabric documentation:

- https://hyperledger-fabric.readthedocs.io/en/release-2.5/couchdb_tutorial.html#cdb-add-index
- https://hyperledger-fabric.readthedocs.io/en/release-2.5/chaincode_lifecycle.html?highlight=chaincode%20package#step-one-packaging-the-smart-contract

**Reply by csb123452010** [reply]
@bestbeforetoday 

 I copied the index from go-version asset-transfer-ledger-quires chaincode to go-version asset-transfer-basic chaincode and java-version asset-transfer-basic chaincode. And I changed the field in the index.

The go-version asset-transfer-basic chaincode with index is OK. However, the java-version asset-transfer-basic chaincode with index was not successful as well.

Do you know what's wrong with my deployment and what caused the difference between two basic chaincode? I think the location is correct. 

And I use scripts in fabric-samples/test-network to do that, so the steps seem no fault.

![索引_basic_go效果](https://github.com/hyperledger/fabric-chaincode-java/assets/97946841/2c3a0b24-3ac9-4597-999c-7b5cd735a3ee)

![索引_basic_java效果](https://github.com/hyperledger/fabric-chaincode-java/assets/97946841/0ca5f87f-48d6-45b1-af53-174fa0fc5114)

**Reply by csb123452010** [reply]
@bestbeforetoday 

I create a new virtual machine with Ubuntu 20 in VMware, the fabric version is 2.5.4, and deploy the asset-private-data/chaincode-java.

There are no index as well.

![索引_private_java效果](https://github.com/hyperledger/fabric-chaincode-java/assets/97946841/d52b9b93-c4f3-46e5-b7e3-eeb4683b5a77)
